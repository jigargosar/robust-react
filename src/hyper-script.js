import h from 'react-hyperscript'
import hh from 'hyperscript-helpers'

export const {div} = hh(h)


export {h}
