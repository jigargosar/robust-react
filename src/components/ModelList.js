import {List, ListItemText, MenuItem} from 'material-ui'
import {map, propOr} from 'ramda'
import {h} from '../hyper-script'

const ModelListItem = ({
  model,
  primaryRenderer = propOr('<primaryRenderer or name not found>', 'name'),
  onClick = () => () => {},
}) =>
  h(MenuItem, {onClick: onClick(model)}, [
    h(ListItemText, {primary: primaryRenderer(model)}),
  ])

const ModelList = ({models, ...other}) =>
  h(List, map(model => h(ModelListItem, {model, ...other}), models))

export {ModelList}
