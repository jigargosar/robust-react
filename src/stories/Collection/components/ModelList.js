import {List, ListItemText, MenuItem} from 'material-ui'
import Props from 'prop-types'
import {h} from '../../../hyper-script'

export const ModelListItem = ({model: {name}}) => {
  return h(MenuItem, [h(ListItemText, name)])
}

const modelShape = {
  name: Props.string.isRequired,
}
ModelListItem.propTypes = modelShape

const ModelList = ({models}) => {
  return h(
    List,
    models.map((model, idx) => h(ModelListItem, {key: idx, model})),
  )
}

ModelList.propTypes = {
  models: Props.arrayOf(Props.shape(modelShape)).isRequired,
}

export {ModelList}
