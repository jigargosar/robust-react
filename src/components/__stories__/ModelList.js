import {action} from '@storybook/addon-actions'
import {storiesOf} from '@storybook/react'
import {centerDecorator} from '../../storybook-helpers'
import {h} from '../../hyper-script-utils'
import {createFakeModels} from '../__fixtures__/Model.fixture'
import {ModelList} from '../ModelList'

const story = storiesOf('Components|Model.List', module).addDecorator(
  centerDecorator,
)

story.add('basic', () =>
  h(ModelList, {
    models: createFakeModels(),
    onClick: model => event => action('onClick')(model, event.type),
  }),
)
story.add('with primaryRenderer', () =>
  h(ModelList, {
    models: createFakeModels(),
    primaryRenderer: model => `${model.name} (id:${model.id})`,
    onClick: model => event => action('onClick')(model, event.type),
  }),
)
