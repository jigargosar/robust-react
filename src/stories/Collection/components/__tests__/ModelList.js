import 'dom-testing-library/extend-expect'
import {render} from 'react-testing-library'
import {h} from '../../../../hyper-script'
import {ModelList} from '../ModelList'

it('should render without crashing', () => {
  const {container} = render(h(ModelList, {models: []}))
  expect(container.firstChild).toMatchSnapshot()
})
