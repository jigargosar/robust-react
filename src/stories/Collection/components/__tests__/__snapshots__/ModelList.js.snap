// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<ul
  class="MuiList-root-1 MuiList-padding-2"
/>
`;
